app/javascript/utils/use-local-storage.ts

```diff
@@ -0,0 +1,22 @@
+import React, { useState, useCallback } from 'react'
+
+export function useLocalStorage(key, initialValue) {
```

Comment:
There are various reasons why not to use localStorage, and [I don't in my preview](https://github.com/SleeplessByte/javascript-preview/blob/d66d506ca9971fe00b1412929ac2c088db40120a/src/state/state.ts), mostly because you'll be hitting limits *fast*. 

You can use [`use-memory-value`](https://www.npmjs.com/package/use-memory-value), which uses whatever storage is available. It's slightly out of date compared to the [`expo-use-memory-value`](https://www.npmjs.com/package/expo-use-memory-value) variant, but I can update it if you'd like.

Reply:
Thanks. I've used this now.

Reply:
@SleeplessByte I seem to encounter a bug where changing the content changes the local storage, but doesn't trigger a render of the whole component. Could you check what I've done wrong with my code? Thanks.

Reply:
Yes. I'll try to do this today. Ping me tonight if I haven't.